downstream.go

```diff
@@ -2074,6 +2077,41 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 			Command: "INVITE",
 			Params:  []string{upstreamUser, upstreamChannel},
 		})
+	case "MONITOR":
+		// Only forward MONITOR commands in single-upstream mode, we don't advertise it otherwise
+		uc := dc.upstream()
+		if uc == nil {
+			return newUnknownCommandError(msg.Command)
+		}
+
+		var subcommand string
+		if err := parseMessageParams(msg, &subcommand); err != nil {
+			return err
+		}
+		monitors := dc.upstream().network.monitors
+		switch subcommand {
+		case "+", "-":
+			var targets string
+			if err := parseMessageParams(msg, nil, &targets); err != nil {
+				return err
+			}
+			switch subcommand {
+			case "+":
+				for _, target := range strings.Split(targets, ",") {
+					monitors.SetValue(target, struct{}{})
+				}
+			case "-":
+				for _, target := range strings.Split(targets, ",") {
+					monitors.Delete(target)
+				}
+			}
+		case "C": // [C]lear the monitors: handle here because no response is returned
+			for k := range monitors.innerMap {
```

Comment:
A fully correct implementation would only clear the downstream conn monitored targets here instead of clearing all targets. I don't care too much.